Dockerfile

```diff
@@ -11,4 +11,24 @@ FROM base as production
 CMD poetry run gunicorn -b "0.0.0.0:80" 'todo_app.app:create_app()'
 
 FROM base as development
-CMD poetry run flask run -h "0.0.0.0" -p 80
\ No newline at end of file
+CMD poetry run flask run -h "0.0.0.0" -p 80
+
+FROM base as test
+# Copy test files
+COPY ./todo_app_tests ./tests
+COPY ./todo_app_tests_e2e ./tests_e2e
```

Comment:
You could COPY these in after installing Firefox, geckodriver and your test dependencies, it would be slightly friendlier to Docker's layer caching.